File: socket_client_udp.py
import socket
import sys

SERVER_IP = socket.gethostname()


def client_program():
    # client_socket = socket.socket()  # instantiate
    try:
        client_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    except socket.error as e:
        print("Error creating socket: %s" % e)
        sys.exit(1)
    foregn_ip = input(
        " This is the domain name or ip address of the server? "
    )  # take input
    foregn_port = int(input("foregn_port? "))
    local_port = int(input("local port? "))
    server_addr = (SERVER_IP, foregn_port)
    client_socket.bind(("", local_port))

    print("\nType 'bye' to end \n")

    message = str(input(" what transactionTime? "))  # take input

    while message.lower().strip() != "bye":
        try:
            client_socket.sendto(message.encode("utf-8"), server_addr)  # send message
        except socket.error as e:
            print("Error sending data: %s" % e)
            sys.exit(1)

        try:
            bytesAddressPair = client_socket.recvfrom(1024)  # receive response
        except socket.error as e:
            print("Error receiving data: %s" % e)
            sys.exit(1)

        msg = bytesAddressPair[0].decode("utf-8")

        print(msg)

        message = input(" time again: ?\n ")  # again take input

    client_socket.close()  # close the connection


if __name__ == "__main__":
    client_program()

File: socket_client_udp.cpp
// g++ socket_client_udp.cpp -std=c++17 -o client && ./client
#include <iostream>
#include <string>
#include <cstring>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>

#define SERVER_IP "127.0.0.1"
#define PORT 5000
#define BUFFER_SIZE 1024

void client_program(int port)
{
    const char *host = SERVER_IP;

    int client_socket = socket(AF_INET, SOCK_DGRAM, 0);
    if (client_socket == -1)
    {
        perror("Error creating socket");
        exit(1);
    }

    sockaddr_in server_addr{};
    server_addr.sin_family = AF_INET;
    server_addr.sin_port = htons(port);

    if (inet_pton(AF_INET, host, &(server_addr.sin_addr)) <= 0)
    {
        perror("Address-related error connecting to server");
        exit(1);
    }

    std::cout << "\nType 'bye' to end\n\n";

    char buffer[BUFFER_SIZE] = {0};
    std::string message;

    std::cout << "What transactionTime? ";
    std::getline(std::cin, message);

    while (message != "bye")
    {
        if (sendto(client_socket, message.c_str(), message.size(), 0, reinterpret_cast<struct sockaddr *>(&server_addr), sizeof(server_addr)) == -1)
        {
            perror("Error sending data");
            exit(1);
        }

        memset(buffer, 0, sizeof(buffer));
        socklen_t server_addr_len = sizeof(server_addr);
        if (recvfrom(client_socket, buffer, BUFFER_SIZE, 0, reinterpret_cast<struct sockaddr *>(&server_addr), &server_addr_len) == -1)
        {
            perror("Error receiving data");
            exit(1);
        }
        std::string data(buffer);

        std::cout << "\nReceived from server: " << data << "\n\n";

        std::cout << "TransactionTime again? ";
        std::getline(std::cin, message);
    }
    if (sendto(client_socket, message.c_str(), message.size(), 0, reinterpret_cast<struct sockaddr *>(&server_addr), sizeof(server_addr)) == -1)
    {
        perror("Error sending data");
        exit(1);
    }

    close(client_socket);
}

int main()
{
    client_program(PORT);
    return 0;
}
